library/spreadsheet.inc.php

```diff
@@ -166,9 +181,11 @@ function real2form($fldval)
         // which must not match any existing template name.
         $new_template_name = form2real($_POST['form_new_template_name']);
         if ($new_template_name != $template_name) {
-            $trow = sqlQuery("SELECT id FROM form_$spreadsheet_form_name WHERE " .
-            "id < 0 AND rownbr = -1 AND colnbr = -1 AND value = '" .
-            real2db($new_template_name) . "'");
+            $trow = sqlQuery(
+                "SELECT id FROM " . escape_table_name('form_' . $spreadsheet_form_name) .
+                " WHERE id < 0 AND rownbr = -1 AND colnbr = -1 AND value = ?",
+                array(real2db($new_template_name))
```

Comment:
Remove the call to real2db(), this looks like a legacy way to try to escape for database. We can also remove the call below and remove this real2db() function entirely near the top of this script (demolition is always fun :) ).

Reply:
I'll remove them. There doesn't seem to be any more references to it from this file. Its also just a container function for `addslashes` so we should be safe to remove it.